ludwig/datasets/goemotions/__init__.py

```diff
@@ -34,4 +34,17 @@ class GoEmotions(UncompressedFileDownloadMixin, MultifileJoinProcessMixin, CSVLo
     """
 
     def __init__(self, cache_dir=DEFAULT_CACHE_LOCATION):
-        super().__init__(dataset_name="goemotions", cache_dir=cache_dir)
\ No newline at end of file
+        super().__init__(dataset_name="goemotions", cache_dir=cache_dir)
+
+    def read_file(self, filetype, filename):
+        if filetype == 'tsv':
+            file_df = pd.read_table(os.path.join(self.raw_dataset_path, filename), header=None)
+        return file_df
```

Comment:
if filetype is not tsv, file_df is not initialized.
If this function is local and used only in this class and you know the files are all tsv you probably don-t need the if, otherwise provide and else.